scvi/model/_destvi.py

```diff
@@ -0,0 +1,269 @@
+        super(DestVI, self).__init__(st_adata)
+        self.module = MRDeconv(
+            n_spots=st_adata.n_obs,
+            n_labels=cell_type_mapping.shape[0],
+            sc_state_dict=sc_state_dict,
+            n_genes=st_adata.n_vars,
+            n_latent=n_latent,
+            n_layers=n_layers,
+            n_hidden=n_hidden,
+            **module_kwargs,
+        )
+        self.cell_type_mapping = cell_type_mapping
+        self._model_summary_string = "DestVI Model"
+        self.init_params_ = self._get_init_params(locals())
+
+    @classmethod
+    def from_rna_model(
+        cls,
+        st_adata: AnnData,
+        sc_model: CondSCVI,
+        **model_kwargs,
+    ):
+        """
+        Alternate constructor for exploiting a pre-trained model on RNA-seq data.
+
+        Parameters
+        ----------
+        st_adata
+            registed anndata object
+        sc_model
+            trained CondSCVI model
+        use_gpu
+            Use the GPU or not.
+        **model_kwargs
+            Keyword args for :class:`~scvi.model.DestVI`
+        """
+        state_dict = (
+            sc_model.module.decoder.state_dict(),
+            sc_model.module.px_decoder.state_dict(),
+            sc_model.module.px_r.detach().cpu().numpy(),
+        )
+
+        return cls(
+            st_adata,
+            sc_model.scvi_setup_dict_["categorical_mappings"]["_scvi_labels"][
+                "mapping"
+            ],
+            state_dict,
+            sc_model.module.n_latent,
+            sc_model.module.n_layers,
+            sc_model.module.n_hidden,
+            **model_kwargs,
+        )
+
+    def get_proportions(
+        self, dataset: AnnData = None, keep_noise: bool = False
+    ) -> np.ndarray:
+        """
+        Returns the estimated cell type proportion for the spatial data.
+
+        Shape is n_cells x n_labels OR n_cells x (n_labels + 1) if keep_noise.
+
+        Parameters
+        ----------
+        dataset
+            the anndata to loop through (used only if amortization)
+        keep_noise
+            whether to account for the noise term as a standalone cell type in the proportion estimate.
+        """
+        column_names = self.cell_type_mapping
+        if keep_noise:
+            column_names = np.append(column_names, "noise_term")
+
+        if self.module.amortization in ["both", "proportion"]:
+            data = dataset.X
+            if isspmatrix(data):
+                data = data.A
+            dl = DataLoader(
+                TensorDataset(torch.tensor(data, dtype=torch.float32)), batch_size=128
+            )  # create your dataloader
+            prop_ = []
+            for tensors in dl:
+                prop_local = self.module.get_proportions(x=tensors[0])
+                prop_ += [prop_local.cpu()]
+            data = np.array(torch.cat(prop_))
+        else:
+            data = self.module.get_proportions(keep_noise=keep_noise)
```

Comment:
ah I see this now, if user passes in a dataset argument you should raise a warning here that it's not being used.